ui/ducks/swaps/swaps.js

```diff
@@ -645,6 +723,90 @@ export const fetchQuotesAndSetQuoteState = (
+  metaMetricsEvent,
+}) => {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const fetchParams = getFetchParams(state);
+    const { metaData, value: swapTokenValue, slippage } = fetchParams;
+    const { sourceTokenInfo = {}, destinationTokenInfo = {} } = metaData;
+    const usedQuote = getUsedQuote(state);
+    const swapsRefreshStates = getSwapsRefreshStates(state);
+    try {
+      dispatch(
+        setSmartTransactionsRefreshInterval(
+          swapsRefreshStates?.stxBatchStatusRefreshTime,
+        ),
+      );
+      // sign and send stx
+      const uuid = await dispatch(
+        signAndSendSmartTransaction({
+          unsignedTransaction,
+          unsignedTransactionsAndEstimates,
+        }),
+      );
+      // update stx with data
+      const destinationValue = calcTokenAmount(
+        usedQuote.destinationAmount,
+        destinationTokenInfo.decimals || 18,
+      ).toPrecision(8);
+      const smartTransactionsOptInStatus = getSmartTransactionsOptInStatus(
+        state,
+      );
+      const smartTransactionsEnabled = getSmartTransactionsEnabled(state);
+      const swapMetaData = {
+        token_from: sourceTokenInfo.symbol,
+        token_from_amount: String(swapTokenValue),
+        token_to: destinationTokenInfo.symbol,
+        token_to_amount: destinationValue,
+        slippage,
+        custom_slippage: slippage !== 2,
+        best_quote_source: getTopQuote(state)?.aggregator,
+        available_quotes: getQuotes(state)?.length,
+        other_quote_selected:
+          usedQuote.aggregator !== getTopQuote(state)?.aggregator,
+        other_quote_selected_source:
+          usedQuote.aggregator === getTopQuote(state)?.aggregator
+            ? ''
+            : usedQuote.aggregator,
+        average_savings: usedQuote.savings?.total,
+        performance_savings: usedQuote.savings?.performance,
+        fee_savings: usedQuote.savings?.fee,
+        median_metamask_fee: usedQuote.savings?.medianMetaMaskFee,
+        stx_enabled: smartTransactionsEnabled,
+        stx_user_opt_in: smartTransactionsOptInStatus,
+      };
+      metaMetricsEvent({
+        event: 'STX Swap Started',
+        category: 'swaps',
+        sensitiveProperties: swapMetaData,
+      });
+      const destinationTokenAddress = destinationTokenInfo.address;
+      const destinationTokenDecimals = destinationTokenInfo.decimals;
+      const destinationTokenSymbol = destinationTokenInfo.symbol;
+      const sourceTokenSymbol = sourceTokenInfo.symbol;
+      const type = TRANSACTION_TYPES.SWAP;
+      await dispatch(
+        updateSmartTransaction(uuid, {
+          origin: 'metamask',
+          destinationTokenAddress,
+          destinationTokenDecimals,
+          destinationTokenSymbol,
+          sourceTokenSymbol,
+          swapMetaData,
+          swapTokenValue,
+          type,
+        }),
+      );
+    } catch (e) {
+      console.log('signAndSendSwapsSmartTransaction error', e);
```

Comment:
We set up a `smartTransactionError` on state for 5xx errors, which should disable smart transactions, but we need to actually simulate this failure and see what happens.

This could be handled by one of the pending items I mentioned: Show a notification to a user when a fallback to regular txs happened